sunpy/net/dataretriever/sources/gong.py

```diff
@@ -0,0 +1,282 @@
+    """
+    Returns a list of URLS to GONG files corresponding to value of input timerange.
+    URL source: `ftp://gong2.nso.edu/QR/` and `http://gong2.nso.edu/HA/haf/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Arguments are any one from ['bb', 'ct', 'le', 'ml', 'td', 'ud','z']
+
+    Physobs: Two arguments or none 'intensity' and 'los_magnetic_field'
+
+    GONGClient expects at least one argument from Physobs or Instrument
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> res = Fido.search(a.Time('2016/6/4', '2016/6/4 00:10:00'), a.Physobs('intensity'))
+    >>> print (res)
+    [<Table length=32>
+     Start Time           End Time      Source     Instrument    
+       str19               str19         str4        str18       
+    ------------------- ------------------- ------ ------------------
+    2016-06-04 00:00:00 2016-06-05 00:00:00   GONG Data not Available
+    2016-06-05 00:00:00 2016-06-06 00:00:00   GONG Data not Available
+    2016-06-06 00:00:00 2016-06-07 00:00:00   GONG Data not Available
+    2016-06-07 00:00:00 2016-06-08 00:00:00   GONG Data not Available
+    2016-06-08 00:00:00 2016-06-09 00:00:00   GONG Data not Available
+    2016-06-09 00:00:00 2016-06-10 00:00:00   GONG Data not Available
+    2016-06-10 00:00:00 2016-06-11 00:00:00   GONG Data not Available
+    2016-06-11 00:00:00 2016-06-12 00:00:00   GONG Data not Available
+    2016-06-12 00:00:00 2016-06-13 00:00:00   GONG Data not Available
+    2016-06-13 00:00:00 2016-06-14 00:00:00   GONG Data not Available
+                    ...                 ...    ...                ...
+    2016-06-26 00:00:00 2016-06-27 00:00:00   GONG Data not Available
+    2016-06-27 00:00:00 2016-06-28 00:00:00   GONG Data not Available
+    2016-06-28 00:00:00 2016-06-29 00:00:00   GONG Data not Available
+    2016-06-29 00:00:00 2016-06-30 00:00:00   GONG Data not Available
+    2016-06-30 00:00:00 2016-07-01 00:00:00   GONG Data not Available
+    2016-07-01 00:00:00 2016-07-02 00:00:00   GONG Data not Available
+    2016-07-02 00:00:00 2016-07-03 00:00:00   GONG Data not Available
+    2016-07-03 00:00:00 2016-07-04 00:00:00   GONG Data not Available
+    2016-07-04 00:00:00 2016-07-05 00:00:00   GONG Data not Available
+    2016-07-05 00:00:00 2016-07-06 00:00:00   GONG Data not Available]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        #TO-DO: Figure out where scraper would and wouldn't work
+        table_physobs = {'INTENSITY' : 'i', 'LOS_MAGNETIC_FIELD': 'b'} #legitimate physical observations.
+        table_instruments = ['bb', 'ct', 'le', 'ml', 'td', 'ud','z'] #For magnetogram and intensity
```

Comment:
instruments name should be full name, these letters mean almost nothing, whereas 'udaipur' gives an idea of where it is.
Please, change them to:
- Big Bear
- Mauna Loa
- Udaipur
- Learmonth
- Cerro Tololo
- Teide
That's what the user should input, you then can have a convertion to the id code you want to be able to obtain the right files.